omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/MockTaskServiceWrapper.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web;
+
+import org.openmrs.module.webservices.helper.TaskServiceWrapper;
+import org.openmrs.scheduler.SchedulerException;
+import org.openmrs.scheduler.TaskDefinition;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MockTaskServiceWrapper extends TaskServiceWrapper {
+	
+	public List<String> registeredTasks = new ArrayList<String>();
+	
+	public List<String> scheduledTasks = new ArrayList<String>();
+	
+	@Override
+	public void saveTaskDefinition(TaskDefinition task) {
+		registeredTasks.add(task.getName());
+	}
+	
+	@Override
+	public void unloadTask(TaskDefinition task) throws SchedulerException {
+		scheduledTasks.remove(task.getName());
+		registeredTasks.remove(task.getName());
+	}
+	
+	@Override
+	public void scheduleTask(TaskDefinition task) throws SchedulerException {
+		boolean contains = false;
+		for (String taskDef : scheduledTasks) {
+			if (taskDef.equals(task.getName())) {
+				contains = true;
+				break;
+			}
+		}
+		if (!contains) {
+			scheduledTasks.add(task.getName());
+		}
+	}
+	
+	@Override
+	public void shutDownTask(TaskDefinition task) throws SchedulerException {
+		for (String taskDef : scheduledTasks) {
+			if (taskDef.equals(task.getName())) {
```

Comment:
Shouldn't this also just call Context.getSchedulerService(). shutDownTask(task)
Same for the others.

Reply:
It is also using this for testing purpose :blush: . So I just mocked it